lib0/src/serde/de.rs

```diff
@@ -0,0 +1,905 @@
+use crate::any::Any;
+use serde::de::value::{MapAccessDeserializer, MapDeserializer, SeqDeserializer};
+use serde::de::{IntoDeserializer, MapAccess, SeqAccess, Visitor};
+use serde::{Deserialize, Deserializer};
+use std::any::type_name;
+use std::collections::HashMap;
+use std::convert::TryInto;
+use std::fmt::{Display, Formatter};
+use thiserror::Error;
+
+pub fn from_any<'de, T: Deserialize<'de>>(any: &'de Any) -> Result<T, AnyDeserializeError> {
```

Comment:
I think that `from_any`/`to_any` could be put behind the traits and be implemented by default for any `T` that implements serde Deserialize.